src/services/ui/src/pages/dashboard/Lists/spas/consts.tsx

```diff
@@ -1,15 +1,37 @@
 import { format } from "date-fns";
 import { removeUnderscoresAndCapitalize } from "@/utils";
 import { OsTableColumn } from "@/components/Opensearch/Table/types";
-import { CMS_READ_ONLY_ROLES, UserRoles } from "shared-types";
+import {
+  CMS_READ_ONLY_ROLES,
+  UserRoles,
+  OsMainSourceItem,
+  stateUserSubStatus,
+} from "shared-types";
+
 import { useGetUser } from "@/api/useGetUser";
+import { PackageCheck } from "shared-utils";
 import {
   renderCellActions,
   renderCellDate,
   renderCellIdLink,
 } from "../renderCells";
 import { BLANK_VALUE } from "@/consts";
 
+export const getStateStatusWithSubStatus = (data: OsMainSourceItem) => {
```

Comment:
This utility may be better kept out of this file and in a utility that is reusable.